[DOCS] Improves description for forecast_stats (#50729)
elastic/elasticsearch: docs/reference/ml/anomaly-detection/apis/get-job-stats.asciidoc

--

`forecasts_stats`.`forecasted_jobs`:::
(long) A value of `0` indicates that forecasts do not exist for this job. A 
value of `1` indicates that at least one forecast exists.

`forecasts_stats`.`memory_bytes`:::
(object) The `avg`, `min`, `max` and `total` memory usage in bytes for forecasts 

> related to this job. Omitted if forecasts do not exist.

Reviewer: I generally like starting the sentence with the conditional part, so folks can stop reading if it doesn't apply to them.
```suggestion
related to this job. If there are no forecasts, this property is omitted.
```

`forecasts_stats`.`records`:::
(object) The `avg`, `min`, `max` and `total` number of model_forecast documents 

> written for forecasts related to this job. Omitted if forecasts do not exist.

Reviewer: ```suggestion
written for forecasts related to this job. If there are no forecasts, this property is omitted.
```

`forecasts_stats`.`processing_time_ms`:::
(object) The `avg`, `min`, `max` and `total` runtime in milliseconds for 

> forecasts related to this job. Omitted if forecasts do not exist.

Reviewer: ```suggestion
forecasts related to this job. If there are no forecasts, this property is omitted.
```

`forecasts_stats`.`status`:::
(object) The count of forecasts by their status. For example: 

> {"finished" : 2, "started" : 1}. Omitted if forecasts do not exist.

Reviewer: ```suggestion
{"finished" : 2, "started" : 1}. If there are no forecasts, this property is omitted.
```
